### PR TITLE
Add: recommanded extensions for this project

### PR DESCRIPTION
**Pull request by wildman777**
For issue #10

Add:
-  vscode/extensions.json
- Recommended extensions for most used extensions
- An unwanted recommendation for deprecated extensions who were once popular or most used.

### CODE REVIEW

**Comment by ArmandBriere** [comment]
image

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		// To contribute with Golang
+		"golang.go",                         		// Go language support
+
+		//To contribute with python
+		"ms-python.python",	                 		//Python
+		"ms-python.black-formatter",         		//To format the code with black
+		"slightc.pip-manager",               		//To install pip-free packages
+		"ms-python.isort",							//To sort the imports
+		"the-compiler.python-tox",					//To run the tests
+		"ms-python.pylint",							//To run the linter
```
Comment (ArmandBriere) [inline .vscode/extensions.json:23]: We need additional config to make this one works with our current `pylintrc` file

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+		//To contribute with docker
+		"ms-azuretools.vscode-docker",       		//Docker support
+		"ms-vscode-remote.remote-containers",		//To run the containers remotely with docker-machine and docker-compose
```
Comment (ArmandBriere) [inline .vscode/extensions.json:27]: We don't need those, we don't have any docker  running during development

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		"ms-azuretools.vscode-docker",       		//Docker support
+		"ms-vscode-remote.remote-containers",		//To run the containers remotely with docker-machine and docker-compose 
+
+		//To contribute with HTML
+		"formulahendry.auto-close-tag",      		//To automatically close the tag when you type a tag name.
+		"formulahendry.auto-rename-tag",     		//To automatically rename the tag when you type a tag name.
+		"cstrap.flask-snippets", 		     		//To add snippets for flask framework in HTML editor.
+		"ecmel.vscode-html-css",			 		//To add snippets for html and css 
+		"zignd.html-css-class-completion",			//To add snippets for html and css classes completion
+		"solnurkarim.html-to-css-autocompletion",	//To add snippets for html to css autocompletion
```
Comment (ArmandBriere) [inline .vscode/extensions.json:35]: All HTML extensions can be removed, those are not mandatory
Those are nice to have for the dev if you want them

File: .vscode/extensions.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		"formulahendry.auto-close-tag",      		//To automatically close the tag when you type a tag name.
+		"formulahendry.auto-rename-tag",     		//To automatically rename the tag when you type a tag name.
+		"cstrap.flask-snippets", 		     		//To add snippets for flask framework in HTML editor.
+		"ecmel.vscode-html-css",			 		//To add snippets for html and css 
+		"zignd.html-css-class-completion",			//To add snippets for html and css classes completion
+		"solnurkarim.html-to-css-autocompletion",	//To add snippets for html to css autocompletion 
+
+		//To contribute with JS/TS
+		"xabikos.JavaScriptSnippets",		  		//To add snippets for JS/TS
+		"ms-vscode.vscode-typescript-next",	  		//To add snippets for TS next
```
Comment (ArmandBriere) [inline .vscode/extensions.json:39]: Same for TS extensions

File: .vscode/extensions.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		//To contribute with JS/TS
+		"xabikos.JavaScriptSnippets",		  		//To add snippets for JS/TS
+		"ms-vscode.vscode-typescript-next",	  		//To add snippets for TS next 
+
+		//To contribute with CSS/SCSS
+		"ritwickdey.LiveServer",			  		//To start a live server 
+		"glenn2223.live-sass",				  		//To format the code with sass 
+		"pranaygp.vscode-css-peek",			  		//To peek the CSS code 
+		"mrezechi3l.css-responsive",		  		//To format the code with responsive CSS
+		"kamikillerto.vscode-colorize",		  		//To colorize the code with syntax highlighting
```
Comment (ArmandBriere) [inline .vscode/extensions.json:46]: same for css extensions

File: .vscode/extensions.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		//To contribute with Angular
+		"johnpapa.Angular2",				  		//To add snippets for Angular 2 and Angular 2 routing (Angular 2 routing is not supported in vscode yet)
+		"Angular.ng-template",				  		//To add snippets for Angular 2 templates (ng-template) and ng-content (content) elements 
+		"alexiv.vscode-angular2-files",				//To add snippets for Angular files (component, directive, service, etc.) and templates (html, css, etc.) 
+		"natewallace.angular2-inline",				//To add snippets for Angular 2 inline templates (<ng-template>) and ng-content (<content>) elements
+		"humao.rest-client",						//To add snippets for RESTful API calls (GET, POST, PUT, DELETE, etc.) 
+
+		//To contribute with Terraform
+		"hashicorp.terraform",			  			//To add snippets for Terraform (provider, resource, etc.) 
+		"4ops.terraform",				  			//To add snippets for Terraform (provider, resource, etc.)
```
Comment (ArmandBriere) [inline .vscode/extensions.json:57]: Does this one add something different from `hashicorp.terraform` ?

File: .vscode/extensions.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		"alexiv.vscode-angular2-files",				//To add snippets for Angular files (component, directive, service, etc.) and templates (html, css, etc.) 
+		"natewallace.angular2-inline",				//To add snippets for Angular 2 inline templates (<ng-template>) and ng-content (<content>) elements
+		"humao.rest-client",						//To add snippets for RESTful API calls (GET, POST, PUT, DELETE, etc.) 
+
+		//To contribute with Terraform
+		"hashicorp.terraform",			  			//To add snippets for Terraform (provider, resource, etc.) 
+		"4ops.terraform",				  			//To add snippets for Terraform (provider, resource, etc.) 
+
+		//Newline at the end of json 
+		"ms-vscode.json-newline-at-end-of-file",	//To add newline at the end of json
```
Comment (ArmandBriere) [inline .vscode/extensions.json:60]: Not found

File: .vscode/extensions.json [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+		"ritwickdey.LiveServer",			  		//To start a live server 
+		"glenn2223.live-sass",				  		//To format the code with sass 
+		"pranaygp.vscode-css-peek",			  		//To peek the CSS code 
+		"mrezechi3l.css-responsive",		  		//To format the code with responsive CSS
+		"kamikillerto.vscode-colorize",		  		//To colorize the code with syntax highlighting 
+
+		//To contribute with Angular
+		"johnpapa.Angular2",				  		//To add snippets for Angular 2 and Angular 2 routing (Angular 2 routing is not supported in vscode yet)
+		"Angular.ng-template",				  		//To add snippets for Angular 2 templates (ng-template) and ng-content (content) elements 
+		"alexiv.vscode-angular2-files",				//To add snippets for Angular files (component, directive, service, etc.) and templates (html, css, etc.)
```
Comment (ArmandBriere) [inline .vscode/extensions.json:51]: Could be removed I think

File: .vscode/extensions.json [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+		"glenn2223.live-sass",				  		//To format the code with sass 
+		"pranaygp.vscode-css-peek",			  		//To peek the CSS code 
+		"mrezechi3l.css-responsive",		  		//To format the code with responsive CSS
+		"kamikillerto.vscode-colorize",		  		//To colorize the code with syntax highlighting 
+
+		//To contribute with Angular
+		"johnpapa.Angular2",				  		//To add snippets for Angular 2 and Angular 2 routing (Angular 2 routing is not supported in vscode yet)
+		"Angular.ng-template",				  		//To add snippets for Angular 2 templates (ng-template) and ng-content (content) elements 
+		"alexiv.vscode-angular2-files",				//To add snippets for Angular files (component, directive, service, etc.) and templates (html, css, etc.) 
+		"natewallace.angular2-inline",				//To add snippets for Angular 2 inline templates (<ng-template>) and ng-content (<content>) elements
```
Comment (ArmandBriere) [inline .vscode/extensions.json:52]: Remove

File: .vscode/extensions.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+		"humao.rest-client",						//To add snippets for RESTful API calls (GET, POST, PUT, DELETE, etc.)
```
Comment (ArmandBriere) [inline .vscode/extensions.json:53]: Remove

File: .vscode/extensions.json [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+		"dbaeumer.vscode-eslint",		 		  	//ESLint
+		"wayou.vscode-todo-highlight",			    //Todo Highlight	
+
+		// To contribute with Golang
+		"golang.go",                         		// Go language support
+
+		//To contribute with python
+		"ms-python.python",	                 		//Python
+		"ms-python.black-formatter",         		//To format the code with black
+		"slightc.pip-manager",               		//To install pip-free packages
```
Comment (ArmandBriere) [inline .vscode/extensions.json:20]: It's good but it's not really related to the project, this can be removed I think

**Review by ArmandBriere — CHANGES_REQUESTED** [review]

**Comment by ArmandBriere** [comment]
Delete all comments, it's not supported in json
https://stackoverflow.com/questions/244777/can-comments-be-used-in-json

File: .vscode/extensions.json [hunk 12]
```diff
@@ -0,0 +1,69 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
+	// Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
+
+	// List of extensions which should be recommended for users of this workspace.
+	"recommendations": [
+		//General recommandations
+		"aaron-bond.better-comments", 	     		//Better Comments
+		"esbenp.prettier-vscode",		     		//Prettier
+		"streetsidesoftware.code-spell-checker",	//Code Spell Checker
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:10]: Ok,  no problem

File: .vscode/extensions.json [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+		"dbaeumer.vscode-eslint",		 		  	//ESLint
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:11]: ok

File: .vscode/extensions.json [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+		"wayou.vscode-todo-highlight",			    //Todo Highlight
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:12]: ok

File: .vscode/extensions.json [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+		"streetsidesoftware.code-spell-checker",	//Code Spell Checker
+		"dbaeumer.vscode-eslint",		 		  	//ESLint
+		"wayou.vscode-todo-highlight",			    //Todo Highlight	
+
+		// To contribute with Golang
+		"golang.go",                         		// Go language support
+
+		//To contribute with python
+		"ms-python.python",	                 		//Python
+		"ms-python.black-formatter",         		//To format the code with black
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:19]: alright

File: .vscode/extensions.json [hunk 16]
```diff
@@ -0,0 +1,69 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
+	// Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
+
+	// List of extensions which should be recommended for users of this workspace.
+	"recommendations": [
+		//General recommandations
+		"aaron-bond.better-comments", 	     		//Better Comments
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:8]: Well I will remove it, if you want to keep it, you can tell me and I will put it again

File: .vscode/extensions.json (hunk 8 above)
Comment (wildman777) [reply .vscode/extensions.json:51]: ok

File: .vscode/extensions.json (hunk 9 above)
Comment (wildman777) [reply .vscode/extensions.json:52]: ok

File: .vscode/extensions.json (hunk 10 above)
Comment (wildman777) [reply .vscode/extensions.json:53]: ok

File: .vscode/extensions.json (hunk 11 above)
Comment (wildman777) [reply .vscode/extensions.json:20]: ok

File: .vscode/extensions.json [hunk 17]
```diff
@@ -0,0 +1,69 @@
+{
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:1]: ok, I will remove those who are not essential

[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:1]: I found the right name for the newline at the end of a file.  I copy-pasted wrongly, it is now fixed.

File: .vscode/extensions.json [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+		"wayou.vscode-todo-highlight",			    //Todo Highlight	
+
+		// To contribute with Golang
+		"golang.go",                         		// Go language support
+
+		//To contribute with python
+		"ms-python.python",	                 		//Python
+		"ms-python.black-formatter",         		//To format the code with black
+		"slightc.pip-manager",               		//To install pip-free packages
+		"ms-python.isort",							//To sort the imports
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:21]: ok

File: .vscode/extensions.json [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+		"the-compiler.python-tox",					//To run the tests
```
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:22]: ok

File: .vscode/extensions.json (hunk 2 above)
Comment (wildman777) [reply .vscode/extensions.json:27]: ok, I will remove it

File: .vscode/extensions.json (hunk 3 above)
Comment (wildman777) [reply .vscode/extensions.json:35]: ok, I will remove them

File: .vscode/extensions.json (hunk 4 above)
Comment (wildman777) [reply .vscode/extensions.json:39]: ok

File: .vscode/extensions.json (hunk 6 above)
Comment (wildman777) [reply .vscode/extensions.json:57]: No, I had added it, but I found out that it doesn't really add much more than what hashicorp.terraform did, so I will remove it

File: .vscode/extensions.json (hunk 7 above)
Comment (wildman777) [reply .vscode/extensions.json:60]: I found the right name for the newline at the end of a file. I copy-pasted wrongly, it is now fixed.

File: .vscode/extensions.json (hunk 17 above)
[the remark this answers is not in the document]
Comment (wildman777) [reply .vscode/extensions.json:1]: You can tell me if there are extension that you want to add in the list.

File: .vscode/extensions.json (hunk 1 above)
Comment (wildman777) [reply .vscode/extensions.json:23]: Ok, since there are config done so that it can work, I will remove it for now

**Review by wildman777 — COMMENTED** [review]
Fix: 

- All changement asked done
- Extension `ms-python.pylint` is removed for now since it need special config
- The correct name of the newline extension: `samverschueren.final-newline`
- All comment in JSON removed

**Pull request merged**